pkg/attestation/crafter/materials/zap.go

```diff
@@ -0,0 +1,129 @@
+//
+// Copyright 2024 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package materials
+
+import (
+	"archive/zip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+
+	schemaapi "github.com/chainloop-dev/chainloop/app/controlplane/api/workflowcontract/v1"
+	"github.com/chainloop-dev/chainloop/internal/casclient"
+	api "github.com/chainloop-dev/chainloop/pkg/attestation/crafter/api/attestation/v1"
+
+	"github.com/rs/zerolog"
+)
+
+const (
+	// zapReportFileName is the name of the file that contains the ZAP report in JSON format
+	zapReportFileName = "report_json.json"
+	// zapProgramName is the name of the program that generated the ZAP report
+	zapProgramName = "ZAP"
+	// outputFileName is the name of the fixed JSON report file
+	outputFileName = "zap_dast_report.json"
+)
+
+// zapJSON is the structure of the ZAP report JSON with the values being checked
+type zapJSON struct {
+	ProgramName string `json:"@programName"`
+}
+
+type ZAPCrafter struct {
+	*crafterCommon
+	backend *casclient.CASBackend
+}
+
+func NewZAPCrafter(materialSchema *schemaapi.CraftingSchema_Material, backend *casclient.CASBackend, l *zerolog.Logger) (*ZAPCrafter, error) {
+	if materialSchema.Type != schemaapi.CraftingSchema_Material_ZAP_DAST_ZIP {
+		return nil, fmt.Errorf("material type is not zap dast zip file")
+	}
+
+	return &ZAPCrafter{
+		backend:       backend,
+		crafterCommon: &crafterCommon{logger: l, input: materialSchema},
+	}, nil
+}
+
+// Craft will extract the ZAP JSON report from the zip file and upload it to the CAS
+func (i *ZAPCrafter) Craft(ctx context.Context, filePath string) (*api.Attestation_Material, error) {
+	archive, err := zip.OpenReader(filePath)
```

Comment:
could we check if it's a zip file?

Reply:
That OpenReader checks already that the incoming file is a zip file but we can do a manual check in any case.